### PR TITLE
Port

### PR DESCRIPTION
**Pull request by fbeutin-ledger**

### CODE REVIEW

File: src/ui/menu_nbgl.c [hunk 1]
```diff
@@ -15,34 +15,25 @@ static void app_quit(void) {
...
 void ui_idle(void) {
-    nbgl_useCaseHome(APPNAME,
-                     &C_icon_exchange_64x64,
-                     "This app enables swapping\nand selling assets\nin Ledger Live.",
-                     true,
-                     ui_menu_settings,
-                     app_quit);
+    nbgl_useCaseHomeAndSettings(APPNAME,
+                                &C_icon_exchange_64x64,
+                                "This app enables swapping\nand selling assets\nin Ledger Live.",
```
Comment (cedelavergne-ledger) [inline src/ui/menu_nbgl.c:31]: `\n` should not be needed

File: src/ui/menu_nbgl.c [hunk 2]
```diff
@@ -15,34 +15,25 @@ static void app_quit(void) {
     os_sched_exit(-1);
 }
 
-static const char* const info_types[] = {"Version", "Exchange App"};
-static const char* const info_contents[] = {APPVERSION, "(c) 2023 Ledger"};
+#define SETTING_INFO_NB 2
+static const char* const INFO_TYPES[SETTING_INFO_NB] = {"Version", "Exchange App"};
+static const char* const INFO_CONTENTS[SETTING_INFO_NB] = {APPVERSION, "(c) 2023 Ledger"};
```
Comment (tdejoigny-ledger) [inline src/ui/menu_nbgl.c:20]: 2024 instead of 2023

File: src/ui/validate_transaction_nbgl.c [hunk 3]
```diff
@@ -151,107 +203,60 @@ static void continue_review(void) {
...
+                        reject_tx);
 }
 
-void display_signing_success(void) {
-    nbgl_useCaseStatus("TRANSACTION\nSIGNED", true, ui_idle);
+static void review_choice(bool confirm) {
+    if (confirm) {
+        accept_tx();
+    } else {
+        rejectUseCaseChoice();
```
Comment (tdejoigny-ledger) [inline src/ui/validate_transaction_nbgl.c:241]: here we could directly call the functions called in `reject_tx` and  then remove `rejectUseCaseChoice `and `reject_tx`

Comment (fbeutin-ledger) [reply src/ui/validate_transaction_nbgl.c]: Done

File: src/ui/menu_nbgl.c (hunk 2 above)
Comment (fbeutin-ledger) [reply src/ui/menu_nbgl.c]: Done

**Pull request merged**